src/test/ship.test.js

```diff
@@ -0,0 +1,48 @@
+import Ship from '../factories/ship';
+
+describe('Ship', () => {
+  let ship;
+
+  beforeEach(() => {
+    ship = Ship(3);
+  });
+
+  describe('hit()', () => {
+    test('The ship is hit nowhere', () => {
+      for(let i=0; i<3; i++) {
+        expect(ship.hits[i]).toBeFalsy();
+      }
+    });
+
+    test('hit(0) set the ship part as hit', () => {
+      ship.hit(0);
+      expect(ship.hits[0]).toBeTruthy();
+    });
+
+    test('hit(5) should raise an error', () => {
+      // expect(() => {
+      //   ship.hit(5);
+      // }).toThrow();
```

Comment:
Consider deleting commented out codes. If you need to review a previous version of your code, use git history.